pkg/alertmanager/clustertlsconfig/config.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2024 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clustertlsconfig
+
+import (
+	"context"
+	"path"
+
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+	webconfig "github.com/prometheus-operator/prometheus-operator/pkg/webconfig"
+)
+
+const (
+	cmdflag            = "cluster.tls-config"
+	volumeName         = "cluster-tls-config"
+	configFile         = "cluster-tls-config.yaml"
+	serverVolumePrefix = "cluster-tls-server-config-"
+	clientVolumePrefix = "cluster-tls-client-config-"
+	serverTLSCredDir   = "server_tls"
+	clientTLSCredDir   = "client_tls"
+)
+
+// Config is the web configuration for prometheus and alertmanager instance.
+//
+// Config can make a secret which holds the web config contents, as well as
+// volumes and volume mounts for referencing the secret and the
+// necessary TLS credentials.
+type ClusterTLSConfig struct {
+	serverTLSConfig     *monitoringv1.WebTLSConfig
+	clientTLSConfig     *monitoringv1.SafeTLSConfig
+	serverTLSReferences *webconfig.TLSReferences
+	clientTLSReferences *webconfig.TLSReferences
+	mountingDir         string
+	secretName          string
+}
+
+// New creates a new ClusterTLSConfig.
+// All volumes related to the cluster TLS config will be mounted via the `mountingDir`.
+// The Secret where the cluster TLS config will be stored will be named `secretName`.
+// All volumes containing TLS credentials related to cluster TLS configuration will be prefixed with "cluster-tls-server-config-"
+// or "cluster-tls-client-config-" respectively, for server and client credentials.
+func New(mountingDir string, secretName string, clusterTLSConfig monitoringv1.ClusterTLSConfigFields) (*ClusterTLSConfig, error) {
```

Comment:
maybe handle nil pointer at this stage?
```suggestion
func New(mountingDir string, secretName string, clusterTLSConfig *monitoringv1.ClusterTLSConfigFields) (*ClusterTLSConfig, error) {
```